cmd/main.go

```diff
@@ -36,7 +45,41 @@ var (
 )
 
 var saramaLogger = log.New(io.Discard, "[Sarama] ", log.LstdFlags)
+
+func InitTracerProvider(ActivateTracing bool, JaegerExporterTracing bool, OtlpExporterTracing bool) *sdktrace.TracerProvider {
+	resources, _ := resource.New(context.Background(),
+		resource.WithFromEnv(), // pull attributes from OTEL_RESOURCE_ATTRIBUTES and OTEL_SERVICE_NAME environment variables
+		resource.WithProcess(), // This option configures a set of Detectors that discover process information
+	)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}))
+	if ActivateTracing {
+		exporter := exporterTracing(JaegerExporterTracing, OtlpExporterTracing)
+		tp := sdktrace.NewTracerProvider(
+			sdktrace.WithResource(resources),
+			sdktrace.WithBatcher(exporter),
+		)
+		otel.SetTracerProvider(tp)
+		return tp
+
+	}
+	otel.SetTracerProvider(trace.NewNoopTracerProvider())
+	return nil
+}
+
+func exporterTracing(JaegerExporterTracing bool, OtlpExporterTracing bool) sdktrace.SpanExporter {
```

Comment:
using booleans sound like if more exporters will be added, we have to change this method signature and getting more parameters. Shouldn't it be much better having just one parameter specifying the exporter to be used? It could be just a string "jaeger" or "otlp" and then the function will create the right exporter accordingly. When adding a new exporter, only the function body will be changed (not in the signature) to create the newly added exporter. In the strimzi bridge we do something like that. We are moving to OTel support and there is a config parameter that you can set to "jaeger" or "otel" and the right exporter will be taken into account.
I see you are building these boolean from the corresponding env vars to be set but it could also returns just a string specifying which exporter we want.

Reply:
i modified it